Source/WebCore/bindings/scripts/test/JS/JSTestObj.cpp

```diff
@@ -2749,6 +2749,9 @@ bool JSTestObj::put(JSCell* cell, JSGlobalObject* lexicalGlobalObject, PropertyN
     auto throwScope = DECLARE_THROW_SCOPE(lexicalGlobalObject->vm());
 
     throwScope.assertNoException();
+    if (!parseIndex(propertyName))
```

Comment:
Here the check isn't redundant though it is not clear to me why we only want to flow through if the property is an index.

Reply:
This is super important because not using `JSObject::put` means "we give up any IC optimization". We should not do that until we actually need to do. And in this case, we should use `JSObject::put`.